src/ddmd/dclass.d

```diff
@@ -967,6 +967,18 @@ extern (C++) class ClassDeclaration : AggregateDeclaration
 
         sc2.pop();
 
+        if (type.ty == Tclass && (cast(TypeClass)type).sym != this)
+        {
+            // https://issues.dlang.org/show_bug.cgi?id=17492
+            ClassDeclaration cd = (cast(TypeClass)type).sym;
+            version (none)
+            {
+                printf("this = %p %s\n", this, this.toPrettyChars());
+                printf("type = %d sym = %p, %s\n", type.ty, cd, cd.toPrettyChars());
+            }
+            error("already exists at %s. Perhaps in another function with the same name?", cd.loc.toChars());
```

Comment:
Maybe split this into an error an errorSupplemental?
```
error("already exists, perhaps in another function with the same name.");
errorSupplemental(cd.loc, "previous declaration found here.");
```
Rather than formating the location into the error string.